### PR TITLE
Курсовая 

### PR DESCRIPTION
**Pull request by SmirnovIlya**
[Review on Reviewable](https://reviewable.io/reviews/basepractice/2015.2/50)

### CODE REVIEW

File: Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/Cell.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                    }
+                    break;
+                }
+                case 1:{
+                    if (!this.left && !to.right) {
+                        return true;
+                    }
+                    break;
+                }
+                case -1:{
```
Comment (Pastor) [inline Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/Cell.java]: Отступ

File: Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/Cell.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+                        case -1:{
+                            if (!this.down && !to.up) {
+                                return true;
+                            }
+                            break;
+                        }
+                    }
+                    break;
+                }
+                case 1:{
```
Comment (Pastor) [inline Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/Cell.java]: Отступ

File: Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/Cell.java [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+            switch (this.x - to.x) {
+                case 0:{
+                    switch (this.y - to.y) {
+                        case 1:{
+                            if (!this.up && !to.down) {
+                                return true;
+                            }
+                            break;
+                        }
+                        case -1:{
```
Comment (Pastor) [inline Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/Cell.java]: Отступ

File: Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/Cell.java [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        if (right) val |= MazeExtension.SQUARE_RIGHT;
+        return val;
+    }
+    public boolean isNeighbor(Cell otherCell) {
+        return Math.abs(this.x - otherCell.x) + Math.abs(this.y - otherCell.y) == 1;
+    }
+    public boolean canMoveTo(Cell to) {
+        if (this.isNeighbor( to)) {
+            switch (this.x - to.x) {
+                case 0:{
```
Comment (Pastor) [inline Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/Cell.java]: Отступ

File: Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/Cell.java [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+        if (left) val |= MazeExtension.SQUARE_LEFT;
+        if (down) val |= MazeExtension.SQUARE_DOWN;
+        if (right) val |= MazeExtension.SQUARE_RIGHT;
+        return val;
+    }
+    public boolean isNeighbor(Cell otherCell) {
+        return Math.abs(this.x - otherCell.x) + Math.abs(this.y - otherCell.y) == 1;
+    }
+    public boolean canMoveTo(Cell to) {
+        if (this.isNeighbor( to)) {
```
Comment (Pastor) [inline Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/Cell.java]: Отступ?

File: Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/MazeEller.java [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+
+    @Override
+    public Point[] findPath(Maze maze) {
+        cols = maze.cols;
+        rows = maze.rows;
+        Cell[][] cells = new Cell[maze.rows][maze.cols];
+        Set<Cell> cellSet = new HashSet<>();
+        for (int y = 0; y < maze.rows; y++) {
+            for (int x = 0; x < maze.cols; x++) {
+                Cell cell=new Cell(y, x, maze.data[y][x]);
```
Comment (Pastor) [inline Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/MazeEller.java]: Отступы

File: Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/MazeEller.java [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+        path.add(finish);
+        Cell currCell=finish;
+        while (!path.contains(start)) {
+            currCell = getNearest(currCell.getNeighbors( cellSet));
+            path.add(currCell);
+        }
+        logger.debug("найден путь от выхода до входа, его длинна "+wave);
+
+        Point[] pathPoints = new Point[path.size()];
+        for (int i=0;i<pathPoints.length;i++) {
```
Comment (Pastor) [inline Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/MazeEller.java]: Отступы

**Comment by Pastor** [comment]
Проверьте отступы и переносы. Исправьте.

**Comment by Pastor** [comment]
Reviewed 7 of 7 files at r1.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/basepractice/2015.2/50)*

**Comment by Pastor** [comment]
Reviewed 7 of 7 files at r1.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/basepractice/2015.2/50)*

**Comment by asm0dey** [comment]
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/MazeEller.java, line 49 \[r1\]](https://reviewable.io:443/reviews/basepractice/2015.2/50#-K5l_jVR8QjTR5eBzRu0:-K5l_jVR8QjTR5eBzRu1:1653764249) ([raw file](https://github.com/basepractice/2015.2/blob/05819a820d63a771bc80158f53c5d314913d8d14/Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/MazeEller.java#L49)):*
`+` должен быть обрамлён пробелами

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/basepractice/2015.2/50)*

**Comment by Pastor** [comment]
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/MazeEller.java, line 49 \[r1\]](https://reviewable.io:443/reviews/basepractice/2015.2/50#-K5l_jVR8QjTR5eBzRu0:-K5la31GBG-1i2xIx1cl:-599003111) ([raw file](https://github.com/basepractice/2015.2/blob/05819a820d63a771bc80158f53c5d314913d8d14/Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/MazeEller.java#L49)):*
А еще лучше в `{}` такой формат поместить :)

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/basepractice/2015.2/50)*

**Comment by asm0dey** [comment]
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/MazeEller.java, line 49 \[r1\]](https://reviewable.io:443/reviews/basepractice/2015.2/50#-K5l_jVR8QjTR5eBzRu0:-K5lbGtwpmhpBdl360Hx:474716807) ([raw file](https://github.com/basepractice/2015.2/blob/05819a820d63a771bc80158f53c5d314913d8d14/Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/MazeEller.java#L49)):*
точно

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/basepractice/2015.2/50)*

**Comment by SmirnovIlya** [comment]
Review status: 5 of 7 files reviewed at latest revision, 8 unresolved discussions.

---

*[Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/MazeEller.java, line 49 \[r1\]](https://reviewable.io:443/reviews/basepractice/2015.2/50#-K5l_jVR8QjTR5eBzRu0:-K5lxx_bxc2a3KFFsHeF:-499146009) ([raw file](https://github.com/basepractice/2015.2/blob/05819a820d63a771bc80158f53c5d314913d8d14/Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/MazeEller.java#L49)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/basepractice/2015.2/50)*

**Comment by SmirnovIlya** [comment]
Review status: 5 of 7 files reviewed at latest revision, 8 unresolved discussions.

---

*[Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/Cell.java, line 80 \[r1\]](https://reviewable.io:443/reviews/basepractice/2015.2/50#-K5kSK7bAfs7XNTxwCVd-r1-80:-K5lyVy9Tq4dKUOMQ_A-:-499146009) ([raw file](https://github.com/basepractice/2015.2/blob/05819a820d63a771bc80158f53c5d314913d8d14/Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/Cell.java#L80)):*
Done.

---

*[Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/Cell.java, line 82 \[r1\]](https://reviewable.io:443/reviews/basepractice/2015.2/50#-K5kSK7bAfs7XNTxwCVd-r1-82:-K5lyXEw2inNxWHt2TzA:-499146009) ([raw file](https://github.com/basepractice/2015.2/blob/05819a820d63a771bc80158f53c5d314913d8d14/Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/Cell.java#L82)):*
Done.

---

*[Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/Cell.java, line 90 \[r1\]](https://reviewable.io:443/reviews/basepractice/2015.2/50#-K5kSK7bAfs7XNTxwCVd-r1-90:-K5lyZQp_D_HTNra_qx5:-499146009) ([raw file](https://github.com/basepractice/2015.2/blob/05819a820d63a771bc80158f53c5d314913d8d14/Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/Cell.java#L90)):*
Done.

---

*[Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/Cell.java, line 99 \[r1\]](https://reviewable.io:443/reviews/basepractice/2015.2/50#-K5kSK7bAfs7XNTxwCVd-r1-99:-K5ly_OEF28pl-sR3kkf:-499146009) ([raw file](https://github.com/basepractice/2015.2/blob/05819a820d63a771bc80158f53c5d314913d8d14/Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/Cell.java#L99)):*
Done.

---

*[Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/Cell.java, line 105 \[r1\]](https://reviewable.io:443/reviews/basepractice/2015.2/50#-K5kSK7bAfs7XNTxwCVd-r1-105:-K5lyaVCaRompE9PHpqA:-499146009) ([raw file](https://github.com/basepractice/2015.2/blob/05819a820d63a771bc80158f53c5d314913d8d14/Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/Cell.java#L105)):*
Done.

---

*[Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/MazeEller.java, line 19 \[r1\]](https://reviewable.io:443/reviews/basepractice/2015.2/50#-K5kSK7bAfs7XNTxwCVe-r1-19:-K5lydDyq3CuIaTaQoBU:-499146009) ([raw file](https://github.com/basepractice/2015.2/blob/05819a820d63a771bc80158f53c5d314913d8d14/Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/MazeEller.java#L19)):*
Done.

---

*[Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/MazeEller.java, line 52 \[r1\]](https://reviewable.io:443/reviews/basepractice/2015.2/50#-K5kSK7bAfs7XNTxwCVe-r1-52:-K5lyeePwcIflrNN0Njo:-499146009) ([raw file](https://github.com/basepractice/2015.2/blob/05819a820d63a771bc80158f53c5d314913d8d14/Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/MazeEller.java#L52)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/basepractice/2015.2/50)*

**Comment by SmirnovIlya** [comment]
Замечания устранил 

---

Review status: 5 of 7 files reviewed at latest revision, 8 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/basepractice/2015.2/50#-:-K5lym85Z6SRZssaPmbu:-1419751960)*

**Comment by Pastor** [comment]
Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/MazeEller.java, line 87 \[r2\]](https://reviewable.io:443/reviews/basepractice/2015.2/50#-K5oUzLtstRUrmwDBuCK:-K5oUzLtstRUrmwDBuCL:-845465410) ([raw file](https://github.com/basepractice/2015.2/blob/094d68f2ce4a2853fea4b516a333e09260492ca1/Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/MazeEller.java#L87)):*
А зачем статические? Да еще и открытые в пакете?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/basepractice/2015.2/50)*

**Comment by SmirnovIlya** [comment]
Review status: 6 of 7 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/MazeEller.java, line 87 \[r2\]](https://reviewable.io:443/reviews/basepractice/2015.2/50#-K5oUzLtstRUrmwDBuCK:-K5rj1uSTLoFINSHQUzA:-499146009) ([raw file](https://github.com/basepractice/2015.2/blob/094d68f2ce4a2853fea4b516a333e09260492ca1/Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/MazeEller.java#L87)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/basepractice/2015.2/50)*

**Comment by Pastor** [comment]
Оформляйтесь

---

Reviewed 4 of 4 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/MazeEller.java, line 125 \[r3\]](https://reviewable.io:443/reviews/basepractice/2015.2/50#-K5u8XcHG8kuMrJgi4E-:-K5u8XcHG8kuMrJgi4E0:1803123751) ([raw file](https://github.com/basepractice/2015.2/blob/9d6d51f58bd4495b0ca1567671161dbe328f51bd/Examples/s131249/src/main/java/ru/mirea/oop/practice/coursej/s131249/MazeEller.java#L125)):*
Для чего метод открытый? По большому счету, у вас должны быть открытыми только интерфейсные методы, остальные закрыты

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/basepractice/2015.2/50#-:-K5u8wZr8y34EbcDE3ZY:-426260396)*

**Pull request merged**